api/src/services/users/create_saved_search.py

```diff
@@ -0,0 +1,18 @@
+import logging
+from uuid import UUID
+
+from src.adapters import db
+from src.db.models.user_models import UserSavedSearch
+
+logger = logging.getLogger(__name__)
+
+
+def create_saved_search(db_session: db.Session, user_id: UUID, json_data: dict) -> UserSavedSearch:
+    saved_search = UserSavedSearch(
+        user_id=user_id, name=json_data["name"], search_query=json_data["search_query"]
+    )
+
+    with db_session.begin():
+        db_session.add(saved_search)
```

Comment:
We can exclude the begin here since we already began it in the route. Actually, in a lot of cases _this_ would error with an "already started transaction error", guessing that didn't happen because you hadn't done any DB operations yet.